example4.py

```diff
@@ -1,24 +1,29 @@
-# Corrected Python code
-
 from math import pi
 
-def calculate_area(radius):
+def calculate_area(radius: float) -> float:
```

Comment:
Adding a check for a negative radius in the `calculate_area` function ensures that the function handles invalid inputs and raises a `ValueError` when necessary.